src/lib.rs

```diff
@@ -179,6 +248,26 @@ fn test_pop_push() {
     assert_eq!([1], *a);
 }
 
+#[test]
+fn test_assigned() {
+    let mut a = vec![0, 1, 2, 3];
+    {
+        assert_eq!([0, 1, 5, 3], *a.assigned(2, 5))
+    }
+    assert_eq!([0, 1, 2, 3], *a);
+}
+
+#[test]
+#[should_panic]
+fn test_assigned_panics_with_out_of_bounds_index() {
+    let mut a = vec![1];
+    {
+        // panics here
+        assert_eq!([1], *a.assigned(2, 5))
+    }
+    assert_eq!([1], *a);
```

Comment:
I don't think either of the assertions in this test get run. I wasn't able to figure out a safe solution for this, but just removing the assertions would be fine with me.

Reply:
@paulkernfeld do you not run into a compile error removing those? i get this https://github.com/paulkernfeld/scoped-ops/pull/1#discussion_r454278980

Reply:
I'm not sure how to reproduce that compile error; maybe it has to do with `Deref` returning `[T]`, which is an unsized type? I tried this:

```rust
#[test]
#[should_panic]
fn test_assigned_panics_with_out_of_bounds_index() {
    vec![1].assigned(2, 5);
}
```

Reply:
ooohh duh. i was doing this:

```rust
let mut a = vec![1];
{
    // panics here
    *a.assigned(2, 5);
}
```

which gives me the error. i still don't understand why, but i'll go with your suggestion, which is much better anyway.